click/simplelog.py

```diff
@@ -0,0 +1,199 @@
+from .core import Option
+import logging
+
+
+def simplelog(*args, **kwargs):
+
+    """ Decorator to add a very simple logging facility to scripts
+
+    For parameters see SimpleLog.__init__
+    """
+
+    def make_simplelog(f):
+
+        return SimpleLog(f, **kwargs)
+
+    return make_simplelog
+
+
+class SimpleLog:
+
+    __name__ = "SimpleLog"
+
+    def __init__(self, callback, verbose_help=None, debug_help=None,
+                 quiet_help=None, remove_args=False, verbose_args=None,
+                 debug_args=None, quiet_args=None, verbose_arg_name=None,
+                 debug_arg_name=None, quiet_arg_name=None,
+                 default_level=None, quiet_level=None, verbose_level=None,
+                 debug_level=None, logger_name=None, flavor=None):
+
+        """ Handler class for SimpleLog
+
+        :param verbose_help: Override the text displayed for the verbose
+                     parameter [Enable verbose logging]
+        :param debug_help: Override the text displayed for the debug parameter
+                           [Enable debugging]
+        :param quiet_help: Override the text displayed for the quiet parameter
+                           [Enable quiet mode]
+        :param remove_args: Remove the logging arguments, so they are not
+                            submitted to the main command [Disabled]
+        :param verbose_args: List of arguments for the "verbose"-setting
+                             [--verbose and -v]
+        :param debug_args: List of arguments for the "debug"-setting
+                           [--debug and -d]
+        :param quiet_args: List of arguments for the "quiet"-setting
+                           [--quiet and -q]
+        :param verbose_arg_name: Name of the "verbose" argument [verbose]
+        :param debug_arg_name: Name of the "debug" argument [debug]
+        :param quiet_arg_name: Name of the "quiet" argument [quiet]
+        :param default_level: Default level [logging.ERROR]
+        :param verbose_level: Level for "verbose"-setting [logging.INFO]
+        :param debug_level: Level for "debug"-setting [logging.DEBUG]
+        :param quiet_level: Level for "quiet"-setting [logging.CRITICAL]
+        :param logger_name: Logger to configure [root-logger]
+        :param flavor: Parameter-flavor to use:
```

Comment:
I got that request in #277. I use a "-q, -v, -d"-flavor but the user there was suggesting a "multiple of v"-variant, so I added the flavor support.

It doesn't do harm IMHO.